### PR TITLE
Fix typos mostly in comments, docs and test names

### PR DESCRIPTION
**Pull request by minho42**
Found some more typos in various files.

### CODE REVIEW

File: Doc/library/importlib.rst [hunk 1]
```diff
@@ -1379,7 +1379,7 @@ an :term:`importer`.
    bytecode file. An empty string represents no optimization, so
    ``/foo/bar/baz.py`` with an *optimization* of ``''`` will result in a
    bytecode path of ``/foo/bar/__pycache__/baz.cpython-32.pyc``. ``None`` causes
-   the interpter's optimization level to be used. Any other value's string
+   the interpreter's optimization level to be used. Any other value's string
    representation being used, so ``/foo/bar/baz.py`` with an *optimization* of
```
Comment (epicfaace) [inline Doc/library/importlib.rst]:
```suggestion
   representation is used, so ``/foo/bar/baz.py`` with an *optimization* of
```

File: Lib/test/test_winreg.py [hunk 2]
```diff
@@ -229,7 +229,7 @@ def test_connect_registry_to_local_machine_works(self):
         h.Close()
         self.assertEqual(h.handle, 0)
 
-    def test_inexistant_remote_registry(self):
+    def test_inexistent_remote_registry(self):
```
Comment (epicfaace) [inline Lib/test/test_winreg.py]:
```suggestion
    def test_nonexistent_remote_registry(self):
```

File: Lib/unittest/test/testmock/testpatch.py [hunk 3]
```diff
@@ -1651,7 +1651,7 @@ def test_patch_imports_lazily(self):
             p1.stop()
         self.assertEqual(squizz.squozz, 3)
 
-    def test_patch_propogrates_exc_on_exit(self):
+    def test_patch_propagrates_exc_on_exit(self):
```
Comment (epicfaace) [inline Lib/unittest/test/testmock/testpatch.py]:
```suggestion
    def test_patch_propagates_exc_on_exit(self):
```

**Review by epicfaace — CHANGES_REQUESTED** [review]

File: Doc/library/importlib.rst (hunk 1 above)
Comment (minho42) [reply Doc/library/importlib.rst]: Thanks for the suggestions @epicfaace,
"Any other value’s string representation being used, so..."
This line is not very clear to me. I think there should be "If" at the beginning as well.
Anyway, I would like to leave this as is if possible, and get the typo bits merged first.

Comment (terryjreedy) [reply Doc/library/importlib.rst]: Accept the suggestion as 'being' is effectively a typo.  With that change, the sentence is coherent, even if not optimal, whereas it it not now.

**Comment by terryjreedy** [comment]
I am going to make and backport the idlelib changes, revert here, and then review.

File: Modules/expat/loadlibrary.c [hunk 4]
```diff
@@ -88,7 +88,7 @@ HMODULE _Expat_LoadLibrary(LPCTSTR filename)
 
   /* Attempt to find LoadLibraryEx() which is only available on Windows 2000
      and above */
-  pLoadLibraryEx = (LOADLIBRARYEX_FN) GetProcAddress(hKernel32, LOADLIBARYEX);
+  pLoadLibraryEx = (LOADLIBRARYEX_FN) GetProcAddress(hKernel32, LOADLIBRARYEX);
```
Comment (terryjreedy) [inline Modules/expat/loadlibrary.c:91]: I verified with recursive grep that these are the only four instances of LOADLIBARYEX in any .c file or any .py file in lib or tools.  Note that the spelling is LOADLIBRARYEX_FN (with 2nd R) on line 51.

**Review by terryjreedy — APPROVED** [review]
All changes look correct and non-controversial.  I checked that the only code change (LOADLIBARYEX to LOADLIBRARYEX) was complete and consistent.

**Comment by terryjreedy** [comment]
I expected the 3.8 backport to fail; good thing I tried it anyway.
If you or anyone makes a 3.7 backport PR, I think the LOADLIBARYEX fix should be skipped in case people have used it in their code.  They should not be surprised until they try 3.8.

**Comment by minho42** [comment]
Thank you @terryjreedy

**Pull request merged**